src/LightingThread.cpp

```diff
@@ -22,19 +22,30 @@ class cReader :
 		BLOCKTYPE * OutputRows = m_BlockTypes;
 		int InputIdx = 0;
 		int OutputIdx = m_ReadingChunkX + m_ReadingChunkZ * cChunkDef::Width * 3;
-		int MaxHeight = std::min(+cChunkDef::Height, m_MaxHeight + 16);  // Need 16 blocks above the highest
-		for (int y = 0; y < MaxHeight; y++)
+		for (size_t i = 0; i != cChunkData::NumSections; ++i)
 		{
-			for (int z = 0; z < cChunkDef::Width; z++)
+			auto * Section = a_ChunkBuffer.GetSection(i);
+			if (Section == nullptr)
 			{
-				a_ChunkBuffer.CopyBlockTypes(OutputRows + OutputIdx * 16, static_cast<size_t>(InputIdx * 16), 16);
-				InputIdx++;
-				OutputIdx += 3;
-			}  // for z
-			// Skip into the next y-level in the 3x3 chunk blob; each level has cChunkDef::Width * 9 rows
-			// We've already walked cChunkDef::Width * 3 in the "for z" cycle, that makes cChunkDef::Width * 6 rows left to skip
-			OutputIdx += cChunkDef::Width * 6;
-		}  // for y
+				// Skip to the next section
+				OutputIdx += 9 * cChunkData::SectionHeight * cChunkDef::Width;
```

Comment:
This seems wrong, I'd say one ` * cChunkDef::Width` is missing.

Reply:
`OutputIdx` is multiplied by `cChunkDef::Width` whenever it's used so this is correct.